stl/inc/xhash

```diff
@@ -1915,6 +1915,116 @@ protected:
         return _List._Getal();
     }
 
+    struct _Multi_equal_check_result {
+        bool _Equal_possible;
+        _Unchecked_const_iterator _Subsequent_first; // only useful if _Equal_possible
+    };
+
+    _NODISCARD _Multi_equal_check_result _Multi_equal_check_equal_range(
+        const _Hash& _Right, _Unchecked_const_iterator _First1) const {
+        // check equal_range of elements matching *_First1 match the equal_range of elements in _Right
+        auto& _Keyval = _Traits::_Kfn(*_First1);
+        // find the start of the matching run in the other container
+        const size_t _Hashval   = _Right._Traitsobj(_Keyval);
+        const size_type _Bucket = _Hashval & _Right._Mask;
+        auto _First2            = _Right._Vec._Mypair._Myval2._Myfirst[_Bucket << 1];
+        if (_First2 == _Right._Unchecked_end()) {
+            // no matching bucket, therefore no matching run
+            return {};
+        }
+
+        const auto _Bucket_hi = _Right._Vec._Mypair._Myval2._Myfirst[(_Bucket << 1) + 1];
+        for (; _Right._Traitsobj(_Traits::_Kfn(*_First2), _Keyval); ++_First2) {
+            // find first matching element in _Right
+            if (_First2 == _Bucket_hi) {
+                return {};
+            }
+        }
+
+        _Unchecked_const_iterator _Left_stop_at;
+        if
+            _CONSTEXPR_IF(_Traits::_Standard) {
+                _Left_stop_at = _Unchecked_end();
+            }
+        else {
+            // check the first elements for equivalence when !_Standard

```

Comment:
Do we have sufficient correctness tests for both `unordered_meow` and `hash_meow`? You're introducing new logic for `!_Standard` so the risk of breaking `hash_meow` is elevated.

Reply:
Yes, the devcrt `Dev11_*****_container_equality` test has a bunch of edge cases tested here.